core/blockchain.go

```diff
@@ -1206,16 +1212,20 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 
 		// Write the block to the chain and get the status.
 		status, err := bc.WriteBlockWithState(block, receipts, state)
+		t3 := time.Now()
```

Comment:
Instead of measuring and updating, perhaps it would be cleaner to update on the fly?

```go
		start = time.Now()
		status, err := bc.WriteBlockWithState(block, receipts, state)
		blockWriteTimer.Update(time.Since(start))
```

Reply:
The only thing it doesn't cover is the timer updates. But my solution does cover bad blocks and failures too, whereas yours only counts successful blocks.

Reply:
Maybe its just the `t0`, `t1`, etc that bother me. Perhaps if they had more meaningful names it would be fine. This feels like some left in debug code., not production.

Reply:
Ah, ok. You want to measure N identifiable and specific things, whereas my approach "we have this chunk of code and want to divvy it up in N places".  So for me `t0, t1.. ` makes sense, but I can see why you dislike it....